hubmap/dataportal/views.py

```diff
@@ -190,3 +220,91 @@ def serialize_multi_dim_counts(data: xr.DataArray):
         pass
 
     return list_for_frontend
+
+
+def globus(request):
+    uuid = None
+    access_token = None
+    refresh_token = None
+    if not hasattr(request, 'user'):
+        request.user = AnonymousUser
+
+    elif request.user.is_authenticated:
+        response = HttpResponseRedirect('http://localhost:3000/')
+        uuid = request.user.social_auth.get(provider='globus').uid
+        social = request.user.social_auth
+        access_token = social.get(provider='globus').extra_data['access_token']
+        refresh_token = social.get(provider='globus').extra_data['refresh_token']
+        response.set_cookie('first_name', request.user.first_name)
+        response.set_cookie('last_name', request.user.last_name)
+        response.set_cookie('email', request.user.email)
+        return response
+    return render(
+        request,
+        'globus.html',
+        {
+            'uuid': uuid,
+            'access_token': access_token,
+            'refresh_token': refresh_token,
+        },
+    )
+
+
+class GlobusUserAuth(generics.GenericAPIView):
+    serializer_class = UserSerializer
+    parser_classes = [JSONParser]
+    versioning_class = versioning.QueryParameterVersioning
+    queryset = User.objects.all()
+
+    def get(self, request, format=None):
+        if not hasattr(request, 'user'):
+            request.user = AnonymousUser
+        if request.user.is_authenticated:
+            # uuid = request.user.social_auth.get(provider='globus').uid
+            # social = request.user.social_auth
+            # access_token = social.get(provider='globus').extra_data['access_token']
+            # refresh_token = social.get(provider='globus').extra_data['refresh_token']
+            response = LoggedInUserSerializer(request.user).data
+        return Response(response)
+
+
+def logout(request):
+    if request.user.is_authenticated:
+        auth_logout(request)
+    response = HttpResponseRedirect(settings.LOGOUT_REDIRECT_URL)
+    response.delete_cookie('first_name')
+    response.delete_cookie('last_name')
+    response.delete_cookie('email')
+    print('logout')
+    return response
+
+
+def login(request):
+    if request.user.is_authenticated:
```

Comment:
Copy and paste from `logout`? This doesn't seem right.

Reply:
@mccalluc please note: this is **not** a copy paste. i am attempting to create an custom login to address some other errors as I stress tested login options. did you see default implementation about globus authentication? so this is picked up from that. as a matter of fact, globus auth config by default does some authentication login/globus and redircts to login/complete tasks. so this is **not** complete.

Reply:
Fixed in https://github.com/hubmapconsortium/hubmap-data-portal/pull/79/commits/1800f23294dacc8a7f07322a13ff33e031e153b4.